lib/graphql/subscription/gitHubAppScmId.graphql

```diff
@@ -0,0 +1,15 @@
+subscription OnGibHubAppScmId {
+  SCMId {
+    id
+    login @required
+    credential {
+      secret @required
+    }
+    provider {
+        apiUrl
+        providerType @matches(re: "github_com_app")
```

Comment:
Why do we need the non-standard `matches` directive here and can't use `provider(providerType = github_com_app)`?

Reply:
@cdupuis I didn't think `@matches` was non-standard, but also, the provider sub-query doesn't take provider type as an argument. If it did, I could use `@required` - is that considered more standard?

Reply:
I'd consider the argument more standard considering all the other subscriptions we use throughout. The fact providerType is not an argument is confusing IMHO.

Reply:
The graphql API was always like that I think - I'll raise an issue to get that changed.

Can we go ahead as is until that's been sorted?